move-vm-backend/tests/move_vm.rs

```diff
@@ -502,3 +509,39 @@ fn dry_run_gas_strategy_doesnt_update_storage() {
         "received module although the dry run strategy was enabled"
     );
 }
+
+#[test]
+// TODO(rqnsom): this one should not panic once the native functions are implemented
+#[should_panic]
+// This test heavily depends on Move.toml files for the used Move packages.
```

Comment:
I'd ask why and how it depends on `Move.toml` files?
Maybe there should be some general comment for all those tests that they depend on `Move.toml` because X, Y, Z..

Reply:
> I'd ask why and how it depends on Move.toml files? 

It's a good question.

We could do something like this (in the future):
Have a module like `assets.rs`, which would have all the info about project locations and their `Move.toml` files.

Then, in our tests, we would use APIs like:
```rust
let module = assets::load_module(MoveModule::UsingStdlibFull_StringAndVector);
let result = vm.publish_module(module.bytecode, module.address, gas);
// e.g. `module.name` would return "StringAndVector"
```

So all those constants like file names, module names, script names, and addresses inside Move. tool would be kept under a structure in the `assets` module.

To choose a module inside our tests we would pick a module with an enum `MoveModule::name_of_the_module`.
The same approach would be used for scripts: `MoveScript::name_of_the_script`
Then all script/module info/data would be available within that predefined struct.

Reply:
For now, I added a general comment on top of the file.